packages/roosterjs-editor-dom/lib/table/VTable.ts

```diff
@@ -120,6 +148,247 @@ export default class VTable {
         );
     }
 
+    /**
+     * Apply the given table format to this virtual table
+     * @param format Table format to apply
+     */
+    applyFormatAndStyle(format: Required<TableFormat>) {
+        if (!this.table) {
+            return;
+        }
+        format = { ...DEFAULT_FORMAT, ...format };
+        this.table.style.borderCollapse = 'collapse';
+        this.setBorderColors(format);
+        this.setTableRowColor(format);
+        this.setTableColumnsColor(format);
+        this.setBorderType(format);
+        this.setFirstColumnFormat(format);
+        this.setHeaderRowFormat(format);
+        this.saveTableInfo(this.table, format);
+    }
+
+    /**
+     * Set color to borders of an table
+     * @param format
+     * @returns
+     */
+    private setBorderColors(format: Partial<TableFormat>) {
+        this.cells.forEach(row =>
+            row
+                .filter(cell => cell.td)
+                .forEach(cell => {
+                    cell.td.style.borderTop = getBorderStyle(format.topBorderColor);
+                    cell.td.style.borderLeft = getBorderStyle(format.verticalBorderColor);
+                    cell.td.style.borderRight = getBorderStyle(format.verticalBorderColor);
+                    cell.td.style.borderBottom = getBorderStyle(format.bottomBorderColor);
+                    cell.td.style.backgroundColor = format.bgColor;
+                })
+        );
+    }
+
+    /**
+     * Set color to even and odd rows
+     * @param format
+     */
+    private setTableRowColor(format: Partial<TableFormat>) {
+        if (!format.bandedRows) {
+            return;
+        }
+        const color = (index: number) => (index % 2 === 0 ? format.bgColorEven : format.bgColorOdd);
+        this.cells.forEach((row, index) => {
+            row.forEach(cell => {
+                const backgroundColor = color(index);
+                if (cell.td && backgroundColor) {
+                    cell.td.style.backgroundColor = backgroundColor;
+                }
+            });
+        });
+    }
+
+    /**
+     * Set color to even and odd columns
+     * @param format
+     */
+    private setTableColumnsColor(format: Required<TableFormat>) {
+        if (!format.bandedColumns) {
+            return;
+        }
+        const color = (index: number) =>
+            index % 2 === 0 ? format.bgColumnColorEven : format.bgColumnColorOdd;
+        this.cells.forEach(row => {
+            row.forEach((cell, index) => {
+                const backgroundColor = color(index);
+                if (cell.td && backgroundColor) {
+                    cell.td.style.backgroundColor = backgroundColor;
+                }
+            });
+        });
+    }
+

```

Comment:
Can these two functions be merged? Both of them are setting background color, if we have two of them, is it possible to have some conflict?